### PR TITLE
Tx replacement improvement

### PR DESCRIPTION
**Pull request by marcindsobczak**
Introduces a minimum requirement of a 10 percent fee increase when replacing transaction

### CODE REVIEW

File: src/Nethermind/Nethermind.TxPool/CompareSameIdentityTxByFee.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            if (ReferenceEquals(x, y)) return 0;
+            if (ReferenceEquals(null, y)) return 1;
+            if (ReferenceEquals(null, x)) return -1;
+            
+            // if gas bottleneck was calculated, it's highest priority for sorting
+            // if not, different method of sorting by gas price is needed
+            if (x.GasBottleneck != 0 || y.GasBottleneck != 0)
+            {
+                y.GasBottleneck.Divide(10, out UInt256 bumpGasBottleneck);
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.TxPool/CompareSameIdentityTxByFee.cs:40]: Could we move 10 to const with the proper name?

File: src/Nethermind/Nethermind.TxPool/CompareSameIdentityTxByFee.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+    public class CompareSameIdentityTxByFee : IComparer<Transaction>
+    {
+        public static readonly CompareSameIdentityTxByFee Instance = new();
+        
+        private CompareSameIdentityTxByFee() { }
+        
+        // To replace old transaction, new transaction needs to have fee higher by at least 10% (1/10) of current fee.
+        // It is required to avoid acceptance and propagation of transaction with almost the same fee as replaced one.
+        private const int PartOfFeeRequiredToIncrease = 10;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.TxPool/CompareSameIdentityTxByFee.cs:32]: Can we move it to TxPoolConfig?

File: src/Nethermind/Nethermind.TxPool/Collections/DistinctValueSortedPool.cs [hunk 3]
```diff
@@ -47,11 +47,14 @@ public abstract class DistinctValueSortedPool<TKey, TValue, TGroupKey> : SortedP
...
         {
-            _comparer = comparer ?? throw new ArgumentNullException(nameof(comparer));
+            // ReSharper disable once VirtualMemberCallInConstructor
+            _comparer = GetSameIdentityComparer(comparer ?? throw new ArgumentNullException(nameof(comparer)));
             _logger = logManager?.GetClassLogger() ?? throw new ArgumentNullException(nameof(logManager));
             _distinctDictionary = new Dictionary<TValue, KeyValuePair<TKey, TValue>>(distinctComparer);
         }
-        
+
+        protected virtual IComparer<TValue> GetSameIdentityComparer(IComparer<TValue> comparer) => comparer;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.TxPool/Collections/DistinctValueSortedPool.cs:56]: Maybe GetReplacementComparer?

File: src/Nethermind/Nethermind.TxPool/CompareSameIdentityTxByFee.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System.Collections.Generic;
+using Nethermind.Core;
+using Nethermind.Int256;
+
+namespace Nethermind.TxPool
+{
+    public class CompareSameIdentityTxByFee : IComparer<Transaction>
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.TxPool/CompareSameIdentityTxByFee.cs:24]: Lets add some docs what the class does

**Review by LukaszRozmej — APPROVED** [review]
Some minor changes would be useful

**Review by tkstanczak — APPROVED** [review]

**Pull request merged**